DataFormats/TrackerCommon/interface/TrackerTopology.h

```diff
@@ -102,6 +102,22 @@ class TrackerTopology {
     unsigned int sterMask_;
   };
 
+  enum DetIdFields {
+    PBModule, PBLadder, PBLayer,
+    PFModule, PFPanel, PFBlade, PFDisk, PFSide, 
+    /* TODO: this can be extended for all subdetectors */
+    DETID_FIELDS_MAX
+  };
+
+ private:
+  struct BitmaskAndSubdet { 
+    unsigned int const& startBit; 
+    unsigned int const& mask;
+    int subdet;
+  };
+  const BitmaskAndSubdet bits_per_field[DETID_FIELDS_MAX];
```

Comment:
better put them in more logical order: the values are derived from pbVals_ etc, it would make more sense to line them out after pbVals_ so that the construction proceeds in logical order.
Also, I think that plain ``` unsigned int const ```  makes more sense: should be faster to access at run time.